oci/layout/oci_dest.go

```diff
@@ -81,6 +91,7 @@ func (d *ociImageDestination) MustMatchRuntimeOS() bool {
 // to any other readers for download using the supplied digest.
 // If stream.Read() at any time, ESPECIALLY at end of input, returns an error, PutBlob MUST 1) fail, and 2) delete any data stored so far.
 func (d *ociImageDestination) PutBlob(stream io.Reader, inputInfo types.BlobInfo) (types.BlobInfo, error) {
+	// This should have been handled during newImageDestination, but just in case
 	if err := ensureDirectoryExists(d.ref.dir); err != nil {
```

Comment:
If `newImageDestination` always creates it, I’d prefer dropping the redundant code. Unless there is some special reason why creating this directory again is actually needed?